bbot/modules/dnscaa.py

```diff
@@ -0,0 +1,84 @@
+# dnscaa.py
+#
+# Checks for and parses CAA DNS TXT records for IODEF reporting destination email addresses and/or URL's.
+#
+# NOTE: when the target domain is initially resolved basic "dns_name_regex" matched targets will be extracted so we do not perform that again here.
+#
+# Example CAA records,
+#   0 iodef "mailto:dnsadmin@example.com"
+#   0 iodef "mailto:contact_pki@example.com"
+#   0 iodef "mailto:ipladmin@example.com"
+#   0 iodef "https://example.com/caa"
+#   0 iodef "https://203.0.113.1/caa" <<< unlikely but possible?
+#   0 iodef "https://[2001:db8::1]/caa" <<< unlikely but possible?
+#
+# We simply extract any URL's as URL_UNVERIFIED, no further activity against URL's is performed by this module in order to remain passive.
+#
+
+from bbot.modules.base import BaseModule
+
+import logging
+
+from bbot.core.helpers.regexes import email_regex, url_regexes
+
+log = logging.getLogger("bbot.core.helpers.dns")
+
+
+class dnscaa(BaseModule):
+    watched_events = ["DNS_NAME"]
+    produced_events = ["EMAIL_ADDRESS", "URL_UNVERIFIED"]
+    flags = ["subdomain-enum", "email-enum", "passive", "safe"]
+    meta = {"description": "Check for CAA iodef records"}
+    options = {
+        "max_event_handlers": 10,
+    }
+    options_desc = {
+        "max_event_handlers": "How many instances of the module to run concurrently",
+    }
+    _max_event_handlers = 10
+
+    async def setup(self):
+        self._max_event_handlers = self.config.get("max_event_handlers", 10)
```

Comment:
This logic is already handled in the [base class](https://github.com/blacklanternsecurity/bbot/blob/1e2a554d024c1886706f931e9de42c1cc135c4f8/bbot/modules/base.py#L336).